### PR TITLE
Re-Re-Re-land implement flutter build bundle with assemble

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

For some reason, if the output file provided to the kernel compiler is missing it fails to complete. Fix this by creating the provided file if missing. Add a test case to handle a null result (exit code != 0) from kernel compiler. Add TODO to fix this in the frontend server

### CODE REVIEW

File: packages/flutter_tools/lib/src/compile.dart [hunk 1]
```diff
@@ -274,6 +274,11 @@ class KernelCompiler {
     if (packagesPath != null) {
       mainUri = PackageUriMapper.findUri(mainPath, packagesPath, fileSystemScheme, fileSystemRoots);
     }
+    // TODO(jonahwilliams): The output file must already exist, but this seems
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/compile.dart]: Maybe it's just the directory of the output file that has to exist already?

**Review by zanderso — APPROVED** [review]
